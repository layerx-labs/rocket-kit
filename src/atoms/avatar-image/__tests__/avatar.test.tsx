import React from 'react';
import { render, screen } from '@testing-library/react';
import 'jest-styled-components';
import { AvatarImage } from '../../..';

describe('Avatar', () => {
  it('renders', () => {
    const { asFragment } = render(
      <AvatarImage alt={'avatar alt'} url={'/dummy.png'} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('has correct alt and src', () => {
    const avatarAlt = 'user avatar';
    render(<AvatarImage alt={avatarAlt} url={'/dummy.png'} />);
    const avatar = screen.queryByAltText(avatarAlt);
    expect(avatar).toHaveProperty('src', 'http://localhost/dummy.png');
  });
});
