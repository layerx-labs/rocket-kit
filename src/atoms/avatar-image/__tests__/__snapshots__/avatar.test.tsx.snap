// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar renders 1`] = `
<DocumentFragment>
  .c0 {
  position: relative;
  border: 0.125rem solid #C7C8D1;
  border-radius: 999px;
  width: 30px;
  height: 30px;
  overflow: hidden;
  z-index: 0;
}

.c0 > * {
  width: 100%;
  height: 100%;
}

.c0 img {
  object-fit: cover;
}

<div
    class="c0 avatar-img"
    size="30"
  >
    <img
      alt="avatar alt"
      src="/dummy.png"
    />
  </div>
</DocumentFragment>
`;
